sdk/monitor/monitor-query/src/internal/modelConverters.ts

```diff
@@ -120,39 +122,27 @@ export function convertResponseForQueryBatch(
  * @internal
  */
 export function fixInvalidBatchQueryResponse(
-  generatedResponse: GeneratedQueryBatchResponse
+  generatedResponse: GeneratedQueryBatchResponse,
+  rawResponse: FullOperationResponse
 ): boolean {
   if (generatedResponse.responses == null) {
     return false;
   }
 
-  let wholeResponse: GeneratedQueryBatchResponse | undefined;
   let hadToFix = false;
 
+  // the body here is incorrect, deserialize the correct one from the raw response itself.
+  const parsedBody = JSON.parse(rawResponse.bodyAsText!);
   // fix whichever responses are in this broken state (each query has it's own
   // response, so they're not all always broken)
   for (let i = 0; i < generatedResponse.responses.length; ++i) {
-    if (
-      generatedResponse.responses[i].body?.tables != null ||
-      generatedResponse.responses[i].body?.error != null
-    ) {
+    if (generatedResponse.responses[i].body?.error != null) {
```

Comment:
@richardpark-msft Do you remember why we had `generatedResponse.responses[i].body?.tables != null` earlier here?

Reply:
Yes.

Reply:
:)

So the problem is that this workaround makes some assumptions about when we're in the error state. What I wanted to be careful about was to not activate if it, somehow, we ended up with real data. So I took the safe route and tried be pretty strict on my check - ie, I only wanted to do it when I was absolutely sure.

It worked when the failure happened in prior iterations but it was depending on how the mistaken deserialization would work, and now that you're migrating to a newer core it might not be the same anymore (ie: we're off the proper path already, so now you just have to adjust it again).

Reply:
Thanks :) Yeah now I need to apply the deserialization logic when we have the data, that means when I have the tables too in the generated response.

Reply:
That's actually very interesting - do you have actual _tables_ or is it just the incorrect deserialized string?

Reply:
The tables didn't have the rows formatted correctly. So using the raw response to deserialize the tables